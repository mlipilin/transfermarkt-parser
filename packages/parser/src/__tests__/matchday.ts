import { matchday } from '../index'

const timeout = Number(process.env.JEST_ASYNC_TIMEOUT)

describe('list', () => {
  test(
    'matchday.list("GB1", "2022") -> list of matchdays for EPL (season 22/23)',
    async () => {
      await expect(matchday.list('GB1', '2022')).resolves.toMatchSnapshot()
    },
    timeout
  )
  test(
    'matchday.list("wrongCompetitionId", "wrongSeasonId") -> []',
    async () => {
      await expect(
        matchday.list('wrongCompetitionId', 'wrongSeasonId')
      ).resolves.toMatchSnapshot()
    },
    timeout
  )
  test(
    'matchday.list() -> []',
    async () => {
      // @ts-ignore
      await expect(matchday.list()).resolves.toMatchSnapshot()
    },
    timeout
  )
})
