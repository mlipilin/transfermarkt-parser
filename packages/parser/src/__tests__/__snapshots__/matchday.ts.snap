// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list matchday.list("GB1", "2022") -> list of matchdays for EPL (season 22/23) 1`] = `
[
  {
    "competitionId": "GB1",
    "id": 1,
    "seasonId": "2022",
    "title": "1.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 2,
    "seasonId": "2022",
    "title": "2.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 3,
    "seasonId": "2022",
    "title": "3.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 4,
    "seasonId": "2022",
    "title": "4.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 5,
    "seasonId": "2022",
    "title": "5.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 6,
    "seasonId": "2022",
    "title": "6.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 7,
    "seasonId": "2022",
    "title": "7.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 8,
    "seasonId": "2022",
    "title": "8.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 9,
    "seasonId": "2022",
    "title": "9.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 10,
    "seasonId": "2022",
    "title": "10.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 11,
    "seasonId": "2022",
    "title": "11.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 12,
    "seasonId": "2022",
    "title": "12.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 13,
    "seasonId": "2022",
    "title": "13.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 14,
    "seasonId": "2022",
    "title": "14.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 15,
    "seasonId": "2022",
    "title": "15.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 16,
    "seasonId": "2022",
    "title": "16.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 17,
    "seasonId": "2022",
    "title": "17.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 18,
    "seasonId": "2022",
    "title": "18.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 19,
    "seasonId": "2022",
    "title": "19.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 20,
    "seasonId": "2022",
    "title": "20.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 21,
    "seasonId": "2022",
    "title": "21.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 22,
    "seasonId": "2022",
    "title": "22.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 23,
    "seasonId": "2022",
    "title": "23.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 24,
    "seasonId": "2022",
    "title": "24.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 25,
    "seasonId": "2022",
    "title": "25.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 26,
    "seasonId": "2022",
    "title": "26.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 27,
    "seasonId": "2022",
    "title": "27.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 28,
    "seasonId": "2022",
    "title": "28.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 29,
    "seasonId": "2022",
    "title": "29.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 30,
    "seasonId": "2022",
    "title": "30.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 31,
    "seasonId": "2022",
    "title": "31.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 32,
    "seasonId": "2022",
    "title": "32.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 33,
    "seasonId": "2022",
    "title": "33.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 34,
    "seasonId": "2022",
    "title": "34.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 35,
    "seasonId": "2022",
    "title": "35.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 36,
    "seasonId": "2022",
    "title": "36.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 37,
    "seasonId": "2022",
    "title": "37.Matchday",
  },
  {
    "competitionId": "GB1",
    "id": 38,
    "seasonId": "2022",
    "title": "38.Matchday",
  },
]
`;

exports[`list matchday.list("wrongCompetitionId", "wrongSeasonId") -> [] 1`] = `[]`;

exports[`list matchday.list() -> [] 1`] = `[]`;
